PG.StarWarsGame.Localisation/PG.StarWarsGame.Localisation/Languages/AlamoLanguageDefinitionBase.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Alamo Engine Tools and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+using System.Globalization;
+using FluentValidation;
+using FluentValidation.Results;
+using PG.Commons.Validation;
+
+namespace PG.StarWarsGame.Localisation.Languages;
+
+/// <summary>
+///     Base definition for an <see cref="IAlamoLanguageDefinition" /> implementation.
+/// </summary>
+public abstract class AlamoLanguageDefinitionBase : IAlamoLanguageDefinition, IValidatable
+{
+    /// <summary>
+    ///     A string that is being used to identify the language of the *.DAT file, e.g. a language identifier
+    ///     "english" would produce the file "mastertextfile_english.dat"
+    /// </summary>
+    protected abstract string ConfiguredLanguageIdentifier { get; }
```

Comment:
Shall this thing be treated case-sensitive or not, concerning equality/hash.

Reply:
Seen as the game doesn't care about capitalization one bit, the definitions shouldn't either.